File: 10170 The Hotel with Infinite Rooms/solution.c
#include <stdio.h>>
#include <math.h>

int main() {
    int S;
    long D;
    while(scanf("%d %ld", &S, &D) == 2)
        printf("%d\n", (int)ceil((-1 + sqrt(1 + ((S * S - S + (D << 1)) << 2))) / 2));
    return 0;
}

File: 10170 The Hotel with Infinite Rooms/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
        StringBuilder output = new StringBuilder();
        while(st.nextToken() == StreamTokenizer.TT_NUMBER) {
            int S = (int)st.nval;
            st.nextToken();
            output.append((int)Math.ceil((-1 + Math.sqrt(1 + ((S * S - S + ((long)st.nval << 1)) << 2))) / 2)).append('\n');
		}
        System.out.print(output);
	}
}
